bokehjs/src/lib/models/comparisons/comparison.ts

```diff
@@ -0,0 +1,24 @@
+import {Model} from "../../model"
+import type * as p from "core/properties"
+
+export namespace Comparison {
+  export type Attrs = p.AttrsOf<Props>
+
+  export type Props = Model.Props
+}
+
+export interface Comparison extends Comparison.Attrs {}
+
+export abstract class Comparison extends Model {
+  declare properties: Comparison.Props
+
+  constructor(attrs?: Partial<Comparison.Attrs>) {
+    super(attrs)
+  }
+
+  override initialize(): void {
+    super.initialize()
+  }
+
+  protected abstract compute(_x: any, _y: any): number
```

Comment:
```suggestion
  protected abstract compute(x: unknown, y: unknown): -1 | 0 | 1
```

1. `abstract` methods don't require `_` prefix to silence unused argument warnings.
2. Always use `unknown` instead of `any`, unless there's a really, really good reason otherwise.
3. The return type is limited in scope, so let's encode that in the type, which will allow its enforcement in subclasses.